src/dc_array.rs

```diff
@@ -154,120 +154,81 @@ impl From<Vec<dc_location>> for dc_array_t {
 }
 
 pub unsafe fn dc_array_unref(array: *mut dc_array_t) {
-    if array.is_null() {
-        return;
-    }
+    assert!(!array.is_null());
     Box::from_raw(array);
 }
 
 pub unsafe fn dc_array_add_uint(array: *mut dc_array_t, item: uintptr_t) {
-    if !array.is_null() {
-        (*array).add_uint(item);
-    }
+    assert!(!array.is_null());
+    (*array).add_uint(item);
 }
 
 pub unsafe fn dc_array_add_id(array: *mut dc_array_t, item: uint32_t) {
-    if !array.is_null() {
-        (*array).add_id(item);
-    }
+    assert!(!array.is_null());
+    (*array).add_id(item);
 }
 
 pub unsafe fn dc_array_add_ptr(array: *mut dc_array_t, item: *mut libc::c_void) {
     dc_array_add_uint(array, item as uintptr_t);
 }
 
 pub unsafe fn dc_array_get_cnt(array: *const dc_array_t) -> size_t {
-    if array.is_null() {
-        0
-    } else {
-        (*array).len()
-    }
+    assert!(!array.is_null());
+    (*array).len()
 }
 
 pub unsafe fn dc_array_get_uint(array: *const dc_array_t, index: size_t) -> uintptr_t {
-    if array.is_null() || index >= (*array).len() {
-        0
-    } else {
-        (*array).get_uint(index)
-    }
+    assert!(!array.is_null());
+    (*array).get_uint(index)
 }
 
 pub unsafe fn dc_array_get_id(array: *const dc_array_t, index: size_t) -> uint32_t {
-    if array.is_null() || index >= (*array).len() {
-        0
-    } else {
-        (*array).get_id(index)
-    }
+    assert!(!array.is_null());
+    (*array).get_id(index)
 }
 
 pub unsafe fn dc_array_get_ptr(array: *const dc_array_t, index: size_t) -> *mut libc::c_void {
-    if array.is_null() || index >= (*array).len() {
-        std::ptr::null_mut()
-    } else {
-        (*array).get_ptr(index)
-    }
+    assert!(!array.is_null());
+    (*array).get_ptr(index)
```

Comment:
Here you remove `index >= (*array).len()` check.

Reply:
`get_ptr()` does this check. It is safe, actually.